transform/cli/cli.go

```diff
@@ -0,0 +1,70 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/konveyor/crane-lib/transform"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type CustomPlugin struct {
+	// TODO: figure out a way to include the name of the plugin in the error messages.
+	name    string
+	runFunc func(*unstructured.Unstructured) (transform.PluginResponse, error)
+}
+
+func (c *CustomPlugin) Run(u *unstructured.Unstructured) (transform.PluginResponse, error) {
+	if c.runFunc == nil {
+		return transform.PluginResponse{}, nil
+	}
+	return c.runFunc(u)
+}
+
+func NewCustomPlugin(name string, runFunc func(*unstructured.Unstructured) (transform.PluginResponse, error)) transform.Plugin {
+	return &CustomPlugin{
+		name:    name,
+		runFunc: runFunc,
+	}
+}
+
+func Unstructured(reader io.Reader) (*unstructured.Unstructured, error) {
+	decoder := json.NewDecoder(reader)
+	u := &unstructured.Unstructured{}
+	err := decoder.Decode(u)
+	return u, err
+}
+
+func ObjectReaderOrDie() io.Reader {
+	return os.Stdin
+}
+
+func StdOut() io.Writer {
```

Comment:
Why do we want to expose this?

Reply:
programmer error. The way this evolved was that I was using Stdout in the main() func but then started writing abstractions around it, never got back to hide everything that was not needed.

Reply:
Updated.